File: app/src/main/java/za/org/grassroot2/view/FragmentView.java
package za.org.grassroot2.view;

import io.reactivex.Observable;

/**
 * Created by luke on 2017/08/25.
 */

public interface FragmentView extends ProgressBarContainer{
    Observable<Integer> viewCreated();
    void showNoConnectionMessage();

    void showErrorDialog(int errorMsgResId);
    void handleNoConnection();
    void handleNoConnectionUpload();
}

File: app/src/main/java/za/org/grassroot2/view/FragmentView.kt
package za.org.grassroot2.view

import io.reactivex.Observable

/**
 * Created by luke on 2017/08/25.
 */

interface FragmentView : ProgressBarContainer {
    fun viewCreated(): Observable<Int>
    fun showNoConnectionMessage()

    fun showErrorDialog(errorMsgResId: Int)
    fun handleNoConnection()
    fun handleNoConnectionUpload()
}
